src/render.js

```diff
@@ -10,7 +11,7 @@ import { createElement, Fragment } from './create-element';
  * render into
  */
 export function render(vnode, parentDom) {
-	let oldVNode = parentDom._prevVNode;
+	let oldVNode = options.checkParent ? options.checkParent(parentDom) : parentDom._prevVNode;
```

Comment:
I think for now we should just add the hook without using it's return value. I can't imagine a scenario where it would make sense to return a completely different dom node.

```js
if (options.root) options.root(parentDom);
let oldVNode = parentDom._prevVNode;
```